Monopoly/Assets/Scripts/Game.cs

```diff
@@ -222,29 +229,115 @@ int calculateNextPlayerIndex(int actualIndex)
+        {
+            currentRent *= GetNumberOfOwnedUtilities(property.GetOwnerId());
+        }
+
+        dialogMenu.ShowForRentPayment(property, playerNames[property.GetOwnerId()], currentRent);
+
+        moneyManager.WithdrawFromAccount(players[payingPlayerId], currentRent);
+        moneyManager.DepositOnAccount(players[property.GetOwnerId()], currentRent);
+
+        currentPlayerBeingPaid = property.GetOwnerId();
+    }
+
+    private int GetNumberOfOwnedStations(int id)
+    {
+        int number = 0;
+        foreach(Property property in players[id].ownedProperties)
+        {
+            if (property.groupName == PropertyGroupName.station)
+                number++;
+        }
+        return number;
+    }
 
+    private int GetNumberOfOwnedUtilities(int id)
+    {
+        int number = 0;
+        foreach (Property property in players[id].ownedProperties)
+        {
+            if (property.groupName == PropertyGroupName.utility)
+                number++;
+        }
+        return number;
     }
+
     void HandleStandingOnOwnPosition(Property property)
     {
-        dialogMenu.ShowForPropertyOwner(property);
+        dialogMenu.ShowForPropertyOwner(property, playerExpandedCurrentProperty, playerDepositedCurrentProperty, ()=> { });
     }
-    void HandleAbleToBuyProperty(Property property, int playerId)
+
+    public void playerExpandedCurrentProperty()
     {
+        moneyManager.WithdrawFromAccount(currentPlayer, currentPlayerStandingProperty.housePrice);
+        if (currentPlayerStandingProperty.IsAbleToBuild() && moneyManager.DoesPlayerHasAnyMoneyLeft(currentPlayer) && !currentPlayerStandingProperty.HasHotel())
+        {
+            currentPlayerStandingProperty.BuildHouse();
+            infoPopup.ShowMessage("", "Rozbudowano nieruchomość");
+        }
+        else
+        {
+            moneyManager.DepositOnAccount(currentPlayer, currentPlayerStandingProperty.housePrice);
+            infoPopup.ShowMessage("", "Nie możesz rozbudować tej nieruchomości");
+        }
+    }
 
-        dialogMenu.ShowAbleToBuy(property,playerBoughtCurrentProperty, () => {  });
-        
+    public void playerDepositedCurrentProperty()
+    {
+        if (currentPlayerStandingProperty.IsDeposited())
+        {
+            moneyManager.WithdrawFromAccount(currentPlayer, currentPlayerStandingProperty.price / 2);
+            if (moneyManager.DoesPlayerHasAnyMoneyLeft(currentPlayer))
+            {
+                currentPlayerStandingProperty.SetDeposit(false);
+                infoPopup.ShowMessage("", "Anulowano zastawienie nieruchomości");
+            }
+            else
+            {
+                moneyManager.DepositOnAccount(currentPlayer, currentPlayerStandingProperty.price / 2);
+                infoPopup.ShowMessage("", "Nie stać Cię na anulowanie zastawienia");
+                currentPlayerStandingProperty.SetDeposit(false);
+                infoPopup.ShowMessage("", "Anulowano zastawienie nieruchomości");
```

Comment:
Tych dwóch ostatnich linijek w bloku else chyba nie powinno tu być? Jeżeli gracz nie ma kasy na odzastawienie, to nie może tego zrobić. Chyba, że źle rozumiem tu kod.

Reply:
Masz rację, to duże niedopatrzenie z mojej strony